initial commit

Lukease opened the pull request.

Commit:
initial commit

jpudysz commented on index.html:
```diff
@@ -0,0 +1,15 @@
+    <script src="node_modules/jquery/dist/jquery.js"></script>
+  </head>
+  <body>
+    <div class='action'></div>
```
double quotes

jpudysz commented on style.css:
```diff
@@ -0,0 +1,125 @@
+body {
```
git ignore

jpudysz commented on main.js:
```diff
@@ -0,0 +1,25 @@
+    const SecondButton = $('<button>').appendTo(buttonsBox).addClass('navigation__buttons').text('Przepisy').attr('disabled', 'disabled')
+    const thirdButton = $('<button>').appendTo(buttonsBox).addClass('navigation__buttons').text('Mishmash').attr('disabled', 'disabled')
+    const addButton = $('<button>').appendTo(action).addClass('navigation__add').text('+')
+    // const productName = $('<div>').appendTo(productsBox).addClass('products__header')
```
remove comments

Commit:
add Events.

Commit:
add css to gitignore.

jpudysz commented on main.js:
```diff
@@ -1,24 +1,83 @@
-        const productText = $('<input>').appendTo(productBox).addClass('selected-product__text').attr('placeholder', 'wpisz nazwe produktu').attr('type', 'text')
-        const trashIcon =  $('<button>').appendTo(productBox).addClass('selected-product__icon-trash')
-        const editIcon =  $('<button>').appendTo(productBox).addClass('selected-product__icon-edit')
+        renderProduct()
```
nice 👌

Commit:
create new function: renderRecipesMenu, addToRecipe, removeFromRecipe,
create new file przepisy_utils.js

jpudysz commented on main.js:
```diff
@@ -12,73 +12,69 @@ const renderMenu = () => {
 
-    products.forEach(object => {
-        renderProduct(object)
+        let products = localStorage.getItem('products')
```
why let?

jpudysz commented on products_utils.js:
```diff
@@ -0,0 +1,76 @@
+    return productBox
+}
+
+export const addAllProducts = addProduct => {
```
proxy

Commit:
add button onClick

jpudysz commented on main.js:
```diff
@@ -1,6 +1,8 @@
 import {renderProduct, addAllProducts} from './products_utils.js'
+import {renderRecipesMenu} from './przepisy_utils.js'
```
use English

jpudysz commented on main.js:
```diff
@@ -1,6 +1,8 @@
 
-const action = $('.action')
+export const action = $('.action')
+let activeButton = 'skladniki'
```
aa

jpudysz commented on main.js:
```diff
@@ -14,67 +16,22 @@ const renderMenu = () => {
-    $('.navigation__add').on('click', () => {
-        renderProduct()
+$('.navigation__buttons:eq(1)').click(() => {
+    activeButton = 'przepisy'
```
aa

jpudysz commented on main.js:
```diff
@@ -14,67 +16,22 @@ const renderMenu = () => {
-
-        return recipes
+$('.navigation__add').click( () => {
+    if (activeButton === 'skladniki') {
```
aa

jpudysz commented on main.js:
```diff
@@ -14,67 +16,22 @@ const renderMenu = () => {
+$('.navigation__add').click( () => {
+    if (activeButton === 'skladniki') {
+        renderProduct()
+        console.log('skladniki')
```
remove console.log

jpudysz commented on main.js:
```diff
@@ -14,67 +16,22 @@ const renderMenu = () => {
-
-secondButton()
-
+    if (activeButton === 'przepisy') {
```
aa

jpudysz commented on main.js:
```diff
@@ -14,67 +16,22 @@ const renderMenu = () => {
-secondButton()
-
+    if (activeButton === 'przepisy') {
+        console.log('przepisy')
```
aa

jpudysz commented on przepisy_utils.js:
```diff
@@ -0,0 +1,59 @@
+}
+
+export const renderRecipesMenu = () => {
+    let recipes = $('<div>').addClass('recipes').appendTo(action)
```
const

Commit:
change names to english.

Commit:
create new function: renderSaveRecipes, addToRecipe, removeFromRecipe,renderRecipes,
create new array: recipesArray, saveArrayOfRecipes, allRecipe,
add new events on: button, trashIcon,product in recipes_utils.js

jpudysz commented on recipes_utils.js:
```diff
@@ -14,46 +32,83 @@ export const removeFromRecipe = event => {
+    const recipesBox = $('.recipe-Box')
+    const readyRecipe = $('<div>').addClass('ready-recipe').appendTo(recipesBox)
+    const trashIcon = $('<button>').appendTo(readyRecipe).addClass('ready-recipe__trash')
+    $('<div>').addClass('ready-recipe__title').appendTo(readyRecipe).text(`name: `).css('font-weight' , 'bold')
```
enter

jpudysz commented on recipes_utils.js:
```diff
@@ -14,46 +32,83 @@ export const removeFromRecipe = event => {
+
+    trashIcon.click(event => {
+        $(event.target).parents('.ready-recipe').remove()
+        const name = $(event.target).siblings('.ready-recipe__name').text()
```
enter before initialization (consts/lets) and instructions

jpudysz commented on recipes_utils.js:
```diff
@@ -14,46 +32,83 @@ export const removeFromRecipe = event => {
+        const name = $(event.target).siblings('.ready-recipe__name').text()
+        allRecipe.forEach((object, index )=> {
+            if (object.recipe.name === name){
+            allRecipe.splice(index, 1)
```
tab

jpudysz replied:
splice mutates array. use slice

jpudysz commented on recipes_utils.js:
```diff
@@ -14,46 +32,83 @@ export const removeFromRecipe = event => {
+            allRecipe.splice(index, 1)
+            }
+        })
+        localStorage.setItem('recipes', JSON.stringify(allRecipe))
```
you can create new file and create a function to wrap localStorage and then you don't need to remember about `JSON.stringify`

jpudysz commented on products_utils.js:
```diff
@@ -19,6 +20,23 @@ const editProduct = event => {
 }
 
+const addLocalStorageToProducts = () => {
+    let saveArrayOfProducts = []
+
+    saveArrayOfProducts = saveArrayOfProducts.concat(localStorage.getItem('products'))
+    saveArrayOfProducts = saveArrayOfProducts.toString().split(',')
+    renderedArrayProducts = saveArrayOfProducts.map(product => {
+        product = product.replaceAll('[', '').replaceAll('"', '').replaceAll(']', '')
+        productsSpecial.add(product)
+
+        return product
+    })
```
You don't need to use let and mutate it:
```
const products = [localStorage.getItem('products').toString().split(',')]
     .map(product => product.replaceAll('[', '').replaceAll('"', '').replaceAll(']', ''))

products.forEach(product => productsSpecial.add(product))
   ```
     
Also you can check comment below and create a helper function like:

```
Storage.get('products')
```
instead of:

```
localStorage.getItem('products').toString().split(',')
```

jpudysz commented on recipes_utils.js:
```diff
@@ -1,6 +1,24 @@
+    const renderedArrayProducts = saveArrayOfRecipes
+
+    renderedArrayProducts.forEach(object => {
+
```
remove enter

jpudysz commented on recipes_utils.js:
```diff
@@ -14,46 +32,83 @@ export const removeFromRecipe = event => {
-            addToRecipe(event)
+    splitProductsName.forEach(productName => {
+        productName = productName.replaceAll('[', '').replaceAll('"', '').replaceAll(']', '')
+        const product = $('<div>').addClass('recipes__products--div').appendTo(recipesProducts).text(productName)
```
enter

jpudysz commented on recipes_utils.js:
```diff
@@ -14,46 +32,83 @@ export const removeFromRecipe = event => {
+            if (!$(event.target).hasClass('recipes__products--div')){
                 removeFromRecipe(event)
-            })
+                return
```
enter before return

jpudysz commented on recipes_utils.js:
```diff
@@ -14,46 +32,83 @@ export const removeFromRecipe = event => {
+                return object.innerHTML
+            })
+
+            const recipe = {name: recipesName, products: array}
```
spaces between `{}`

jpudysz commented on recipes_utils.js:
```diff
@@ -14,46 +32,83 @@ export const removeFromRecipe = event => {
+        if (recipesName !== '' && $('.recipes__products').children().hasClass('recipes__products--select')) {
+            const arrayOfProducts = $('.recipes__products--select').toArray()
+
+            const array = arrayOfProducts.map(object => {
```
```
const array = arrayOfProducts.map(object => {
     return object.innerHTML
 })
```
```
const array = arrayOfProducts.map(object => object.innerHTML)
```

jpudysz commented on main.js:
```diff
@@ -1,35 +1,73 @@
+    const renderedArrayProducts = saveArrayOfProducts.toString().split(',')
+
+    renderedArrayProducts.forEach(object => {
+            object = object.replaceAll('[', '').replaceAll('"', '').replaceAll(']', '')
```
bad indentation

jpudysz commented on main.js:
```diff
@@ -1,35 +1,73 @@
     const firstButton = $('<button>').appendTo(buttonsBox).addClass('navigation__buttons').text('Składniki')
-    const SecondButton = $('<button>').appendTo(buttonsBox).addClass('navigation__buttons').text('Przepisy').attr('disabled', 'disabled')
-    const thirdButton = $('<button>').appendTo(buttonsBox).addClass('navigation__buttons').text('Mishmash').attr('disabled', 'disabled')
+    let SecondButton = $('<button>').appendTo(buttonsBox).addClass('navigation__buttons').text('Przepisy').attr('disabled', 'disabled')
```
variables should be lowercased

Commit:
create new function: renderMishMashChoice, renderCorrectRecipes, findCorrectRecipe
create new array: selectedProducts, recipes,
add new events on: newProduct in mish_mash_utils.js

jpudysz commented on main.js:
```diff
@@ -1,19 +1,21 @@
-            object = object.replaceAll('[', '').replaceAll('"', '').replaceAll(']', '')
-            renderProduct(object)
-            $('.selected-product__text').attr('readOnly', true).css('text-align', 'center')
+        object = object.replaceAll('[', '').replaceAll('"', '').replaceAll(']', '')
```
you should get array from localStorage.

jpudysz commented on products_utils.js:
```diff
@@ -17,20 +24,18 @@ const editProduct = event => {
-        product = product.replaceAll('[', '').replaceAll('"', '').replaceAll(']', '')
-        productsSpecial.add(product)
+    const products = getFromLocalStorage('products')
+        .map(product => product.replaceAll('[', '').replaceAll('"', '').replaceAll(']', ''))
```
aa

jpudysz commented on mish_mash_utils.js:
```diff
@@ -0,0 +1,66 @@
+
+        newProduct.click(event => {
+            $('.recipe-Box').children('.ready-recipe').remove()
+            if (!$(event.target).hasClass('recipes__products--div')) {
```
enter

Commit:
add enter, remove unnecessary code.

jpudysz commented on mish_mash_utils.js:
```diff
@@ -33,15 +33,14 @@ export const renderMishMashChoice = () => {
 
-
-    let splitProductsName = productsName.toString().split(',')
+    let splitProductsName = productsName
```
why? are you sure you need to copy it?

Commit:
remove unnecessary code and array.
